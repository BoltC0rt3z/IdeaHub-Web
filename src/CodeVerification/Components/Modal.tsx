import React from 'react';
import { Row, Col } from 'antd';
import { ModalProps } from 'antd/lib/modal';

import { Modal, Button, Input } from '../../Components';

import styles from './Modal.module.scss';

const VerificationModal: React.FC<ModalProps> = (props) => {
  const { onOk, onCancel } = props;

  return (
    <Modal title="Add email address" footer={null} {...props}>
      <Row>
        <Input placeholder="New email address" />
      </Row>
      <Row type="flex" style={{ flexDirection: 'row-reverse', paddingTop: '32px' }} gutter={8}>
        <Col>
          <Button type="primary" onClick={onOk}>
            Add
          </Button>
        </Col>
        <Col>
          <Button className={styles.button} onClick={onCancel}>
            Cancel
          </Button>
        </Col>
      </Row>
    </Modal>
  );
};

export default VerificationModal;
