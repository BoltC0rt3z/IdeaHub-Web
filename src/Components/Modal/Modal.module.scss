@import '../../styles/styles.module.scss';

.modal {
  :global {
    .ant-modal-title {
      color: $color-dark-blue-lighter;
    }

    .ant-modal-header,
    .ant-modal-footer {
      background: $color-light-gray;
    }
  }
}
