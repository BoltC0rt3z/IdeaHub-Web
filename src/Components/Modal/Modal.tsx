import React from 'react';
import { Modal as AntdModal } from 'antd';
import { ModalProps } from 'antd/lib/modal';

import styles from './Modal.module.scss';

const Modal: React.FC<ModalProps> = (props) => {
  const { children, className, ...restProps } = props;
  const classNames: string = [styles.modal, className].join(' ');

  return (
    <AntdModal className={classNames} {...restProps}>
      {children}
    </AntdModal>
  );
};

export default Modal;
