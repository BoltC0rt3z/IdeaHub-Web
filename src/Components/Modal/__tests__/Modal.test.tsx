import React from 'react';
import { mount, ReactWrapper } from 'enzyme';

import Modal from '../Modal';

describe('Tests for modals', () => {
  const wrapper: ReactWrapper = mount(<Modal />);

  it('Should render the modal without crushing', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
