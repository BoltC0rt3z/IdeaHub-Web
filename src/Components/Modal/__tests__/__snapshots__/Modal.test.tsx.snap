// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for modals Should render the modal without crushing 1`] = `ReactWrapper {}`;
